source/docs/fancy_logger.md

```diff
@@ -0,0 +1,63 @@
+## Fancy Logger: Flexible Granularity Log Control in Envoy
+
+### Overview
+Fancy Logger is a logger with finer grained log level control and runtime logger update using administration interface. Compared to the existing logger in Envoy, Fancy Logger provides file level control which can be easily extended to finer grained logger with function or line level control, and it is completely automatic and never requires developers to explicitly specify the logging component. Besides, it has a comparable speed as Envoy's logger. 
+
+### Basic Usage
+The basic usage of Fancy Logger is to explicitly call its macros:
+```
+  FANCY_LOG(info, "Hello world! Here's a line of fancy log!");
+  FANCY_LOG(error, "Fancy Error! Here's the second message!");
+```
+If the level of log message is higher than that of the file, macros above will print messages with the file name like this:
+```
+[2020-07-29 22:27:02.594][15][error][test/common/common/log_macros_test.cc] [test/common/common/log_macros_test.cc:149] Fancy Error! Here\'s the second message!
+```
+More macros with connection and stream information:
+```
+  NiceMock<Network::MockConnection> connection_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> stream_;
+  FANCY_CONN_LOG(warn, "Fake info {} of connection", connection_, 1);
+  FANCY_STREAM_LOG(warn, "Fake warning {} of stream", stream_, 1);
+```
+To flush a logger, `FANCY_FLUSH_LOG()` can be used. 
+
+### Logger Mode: Fancy or Envoy [Deprecated: commandline option now]
```

Comment:
We don't have any usage of this, just delete this paragraph and let folks use command line option?

Reply:
Addressed now. I think the paragraph should be kept as there's still something to remind user.